tox.ini

```diff
@@ -1,13 +1,36 @@
 [tox]
 envlist =
     clean,
-    # check,
     docs,
     py36-cover,
+    py36-cover-ml
     py36-nocov,
+    py36-nocov-ml,
     py37-cover,
+    py37-cover-ml,
     py37-nocov,
+    py37-nocov-ml,
     report
+    mypy
+
+# ------------- environment setup -----------------
+
+[standarddeps]
+deps =
+    pytest
+    pytest-travis-fold
+    pytest-xdist
+
+[mldeps]
+deps =
+    {[standarddeps]deps}
+    tensorflow
+    https://download.pytorch.org/whl/cpu/torch-1.1.0-cp36-cp36m-win_amd64.whl ; sys_platform == 'win32'
+    https://download.pytorch.org/whl/cpu/torchvision-0.3.0-cp36-cp36m-win_amd64.whl ; sys_platform == 'win32'
+    torch ; sys_platform == 'darwin'
+    torchvision ; sys_platform == 'darwin'
```

Comment:
@rlizzo Why are we doing `torchvision` here. I don't think we need that anywhere?